docs/SUMMARY.md

```diff
@@ -20,6 +20,7 @@
     - [Simulation Configuration]()
         - [Shadow Config Overview](shadow_config_overview.md)
         - [Shadow Config Specification](shadow_config_spec.md)
+        - [Programmatic Config Generation](shadow_config_dynamic.md)
```

Comment:
Since this file contains some stuff that isn't "programmatic" such as yaml anchors, maybe something like "Advanced Config Generation" would be better?

Reply:
I think "advanced" makes it sound like advanced configuration options in Shadow, rather than advanced usage and generation of YAML.

Maybe "Managing Complex Configurations"?

Reply:
Sure, any of the above sound good to me.